File: solid/max_miguel/typescript/EmpleadoPorHoras.ts
import Empleado from "./Empleado";

export default class EmpleadoPorHoras extends Empleado {
  private horasTrabajadas: number;
  private montoPorHora: number;

  constructor(
    horasTrabajadas: number,
    montoPorHora: number,
    nombre: string,
    id: number
  ) {
    super(nombre, id);
    this.horasTrabajadas = horasTrabajadas;
    this.montoPorHora = montoPorHora;
  }

  public calcularSueldo(): number {
    return this.horasTrabajadas * this.montoPorHora;
  }
}

File: max_miguel/python/empleado_por_horas.py
from empleado import Empleado

class EmpleadoPorHoras(Empleado):
    def __init__(self, nombre: str , id: int, horas_trabajadas: int, salario_por_hora: int):
        super().__init__(nombre, id)
        self._horas_trabajadas = horas_trabajadas
        self._salario_por_hora = salario_por_hora

    def calcular_salario(self):
        return self._salario_por_hora * self._horas_trabajadas